### PR TITLE
[persist] Make the streaming consolidate impls the default

### PR DESCRIPTION
**Pull request by bkirwi**
These streaming impls have been enabled in our staging and sandbox envs for a long time with no issues.

### Motivation

We don't expect (or observe) huge improvements from making these streaming, since they both read all the data into memory. (They are _probably_ a bit more memory-efficient, but it's hard to measure.) However, this does let us avoid duplicating a bunch of code for the streaming and non-streaming cases.

### Checklist

- [x] This PR has adequate test coverage / QA involvement has been duly considered.
- [x] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [ ] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [ ] If this PR will require changes to cloud orchestration or tests, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [ ] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  -

### CODE REVIEW

File: src/persist-client/src/internal/compact.rs
Comment (bkirwi) [inline src/persist-client/src/internal/compact.rs:1304]: We have a similar test for the streaming consolidator already, so I don't feel bad about deleting this one.

**Review by danhhz — APPROVED** [review]
\o/

Given that it looks like we only turned these on for the rest of prod yesterday, I'd personally be inclined to wait to merge this until after the release candidate is cut this week just to give us a _bit_ more time to react if prod finds something that the canaries didn't. But also 100% your call, I'm on board if you want to merge it today

**Comment by bkirwi** [comment]
> Given that it looks like we only turned these on for the rest of prod yesterday, I'd personally be inclined to wait to merge this until after the release candidate is cut this week just to give us a bit more time to react if prod finds something that the canaries didn't.

Yep agreed!

**Pull request merged**
